### PR TITLE
Refactor 0x.js

### PR DESCRIPTION
**Pull request by fabioberger**
## Description

As the title says 😄 

## Testing instructions

Run the 0x.js tests

## Types of changes

*   [ ] Bug fix (non-breaking change which fixes an issue)
*   [ ] New feature (non-breaking change which adds functionality)
*   [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

*   [ ] Change requires a change to the documentation.
*   [ ] Added tests to cover my changes.
*   [x] Added new entries to the relevant CHANGELOG.jsons.
*   [ ] Labeled this PR with the 'WIP' label if it is a work in progress.
*   [ ] Labeled this PR with the labels corresponding to the changed package.

### CODE REVIEW

File: packages/contract-wrappers/src/schemas/contract_wrappers_public_network_config_schema.ts [hunk 1]
```diff
@@ -22,8 +22,7 @@ export const contractWrappersPublicNetworkConfigSchema = {
         gasPrice: { $ref: '/Number' },
         zrxContractAddress: { $ref: '/Address' },
         exchangeContractAddress: { $ref: '/Address' },
-        tokenRegistryContractAddress: { $ref: '/Address' },
-        tokenTransferProxyContractAddress: { $ref: '/Address' },
+        erc20ProxyContractAddress: { $ref: '/Address' },
```
Comment (LogvinovLeon) [inline packages/contract-wrappers/src/schemas/contract_wrappers_public_network_config_schema.ts]: What about erc721ProxyContractAddress

File: packages/0x.js/package.json [hunk 2]
```diff
@@ -29,15 +30,15 @@
...
+        "build:commonjs": "tsc && yarn copy_artifacts && copyfiles -u 3 './lib/src/monorepo_scripts/**/*' ./scripts",
         "run_mocha": "mocha --require source-map-support/register --require make-promises-safe lib/test/**/*_test.js lib/test/global_hooks.js --timeout 10000 --bail --exit",
         "manual:postpublish": "yarn build; node ./scripts/postpublish.js",
         "docs:stage": "node scripts/stage_docs.js",
         "docs:json": "typedoc --excludePrivate --excludeExternals --target ES5 --json $JSON_FILE_PATH $PROJECT_FILES",
         "upload_docs_json": "aws s3 cp generated_docs/index.json $S3_URL --profile 0xproject --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers --content-type application/json"
     },
     "config": {
-        "compact_artifacts": "Exchange DummyToken ZRXToken Token EtherToken TokenTransferProxy TokenRegistry",
+        "contracts": "ZRXToken",
```
[the remark this answers is not in the document]
Comment (fabioberger) [reply packages/0x.js/package.json]: It's the only one we use in the tests.

File: packages/0x.js/src/0x.ts [hunk 3]
```diff
@@ -93,7 +87,7 @@ export class ZeroEx {
      * @return  Whether the supplied orderHash has the expected format.
      */
     public static isValidOrderHash(orderHash: string): boolean {
-        return isValidOrderHash(orderHash);
+        return orderHashUtils.isValidOrderHash(orderHash);
```
[the remark this answers is not in the document]
Comment (fabioberger) [reply packages/0x.js/src/0x.ts]: Already exist in OrderHashUtils methods.

File: packages/0x.js/src/0x.ts [hunk 4]
```diff
@@ -66,24 +71,13 @@ export class ZeroEx {
...
-        return isValidSignature(data, signature, signerAddress);
-    }
     /**
      * Computes the orderHash for a supplied order.
      * @param   order   An object that conforms to the Order or SignedOrder interface definitions.
      * @return  The resulting orderHash from hashing the supplied order.
      */
     public static getOrderHashHex(order: Order | SignedOrder): string {
-        return getOrderHashHex(order);
+        return orderHashUtils.getOrderHashHex(order);
```
[the remark this answers is not in the document]
Comment (fabioberger) [reply packages/0x.js/src/0x.ts]: Already exist in OrderHashUtils methods.

**Pull request merged**